pyqtorch/utils.py

```diff
@@ -116,3 +116,20 @@ def _rand(n_qubits: int) -> torch.Tensor:
 
 def param_dict(keys: Sequence[str], values: Sequence[torch.Tensor]) -> dict[str, torch.Tensor]:
     return {key: val for key, val in zip(keys, values)}
+
+
+def density_mat(
+    vector_state: torch.Tensor, dtype: torch.dtype = DEFAULT_MATRIX_DTYPE
+) -> torch.Tensor:
+    """
+    Transforms a pure state :math:`|\\psi\\rangle` into its corresponding density matrix
```

Comment:
Not sure the docstring is formatted correctly.

Reply:
Looking good. Two points here: 1) I would import torch `Tensor` types and use them throughout as opposed to `torch.Tensor` everywhere and 2) rename `vector_state` -> `state` as it is unambiguous that we work here with vectors.